src/apiutils.jl

```diff
@@ -55,36 +55,30 @@ end
 
 function seed!(duals::AbstractArray{Dual{T,V,N}}, x,
                seed::Partials{N,V} = zero(Partials{N,V})) where {T,V,N}
-    for i in eachindex(duals)
-        duals[i] = Dual{T,V,N}(x[i], seed)
-    end
+    duals .= Dual{T,V,N}.(x, Ref(seed))
     return duals
 end
 
 function seed!(duals::AbstractArray{Dual{T,V,N}}, x,
                seeds::NTuple{N,Partials{N,V}}) where {T,V,N}
-    for i in 1:N
-        duals[i] = Dual{T,V,N}(x[i], seeds[i])
-    end
+    i = 1:N
+    duals[i] .= Dual{T,V,N}.(view(x,i), seeds)
     return duals
 end
 
 function seed!(duals::AbstractArray{Dual{T,V,N}}, x, index,
                seed::Partials{N,V} = zero(Partials{N,V})) where {T,V,N}
     offset = index - 1
-    for i in 1:N
-        j = i + offset
-        duals[j] = Dual{T,V,N}(x[j], seed)
-    end
+    j = (1:N) .+ offset
+    duals[j] .= Dual{T,V,N}.(view(x, j), Ref(seed))
     return duals
 end
 
 function seed!(duals::AbstractArray{Dual{T,V,N}}, x, index,
                seeds::NTuple{N,Partials{N,V}}, chunksize = N) where {T,V,N}
     offset = index - 1
-    for i in 1:chunksize
-        j = i + offset
-        duals[j] = Dual{T,V,N}(x[j], seeds[i])
-    end
+    i = 1:chunksize
+    j = i .+ offset
+    duals[j] .= Dual{T,V,N}.(view(x, j), seeds[i])
```

Comment:
I would be tempted to write `seed_inds` and `dual_inds` rather than `i` and `j` because the single letter makes me think it is a scalar.
Similar above where there is 1 variable i would be tempted to call it `inds` rather than `i` or `j` for same reason.

or `irange` like used below